nixos/modules/services/networking/adguardhome.nix

```diff
@@ -41,6 +41,20 @@ in
       '';
     };
 
+    allowDHCP = mkOption {
+      default = attrByPath [ "settings" "dhcp" "enabled" ] false cfg;
```

Comment:
Why not use cfg.settings.dhcp.enabled?

Reply:
`settings` might be `null` or an empty attrset, so we can't be sure whether that path exists.

Reply:
```suggestion
      default = cfg.settings.dhcp.enabled or false;
```

that should work

```
nix-repl> lib.some.thing.random or 1
1
```

Reply:
That appears to be working fine, tests remain passing with this change. I've updated `defaultText` accordingly.